### PR TITLE
refactor: cache handling and improve cache action logs

### PR DESCRIPTION
**Pull request by penandlim**
Cache was not saving on some of my actions with `cache` input not explicitly set - moving the check of `cache` input into js seems to have fixed it.

* Refactor cache handling and improve cache action logs
* Use `getBooleanInput` to ensure boolean type
* Add helpful logging similar to actions/setup-* repos to see which cache was restored/saved.
* [Add early exit on post-action for a workaround.](https://github.com/actions/setup-node/issues/878) With a 200kb cache, post-action time is decreased from 2min~ -> 2 seconds.

### CODE REVIEW

File: src/cache.js [hunk 1]
```diff
@@ -9,16 +10,27 @@ const PLATFORM = os.platform();
...
-  if (fs.existsSync(CACHE_PATHS[0])) {
-    await cache.saveCache(CACHE_PATHS, key);
+  const primaryKey = PLATFORM + "-foundry-chain-fork-" + github.context.sha;
+  if (!fs.existsSync(CACHE_PATHS[0])) {
+    core.info(`Cache path does not exist, not saving cache : ${CACHE_PATHS[0]}`);
+    return;
+  }
+  const cacheId = await cache.saveCache(CACHE_PATHS, primaryKey);
+  if (cacheId === -1) {
+    return;
```
Comment (onbjerg) [inline src/cache.js:31]: when does this happen? should we add a log?

Comment (penandlim) [reply src/cache.js:31]: Looks like it happens when there is an existing cache with the primary key thus fails saving the cache. 
It doesn't throw an error and prints a message like this before returning with -1.

```
Failed to save: Unable to reserve cache with key linux-foundry-chain-fork-c2ec5babbad85fecbae9c860472722f98f36c164, another job may be creating this cache. More details: Cache already exists. Scope: refs/heads/master, Key: linux-foundry-chain-fork-c2ec5babbad85fecbae9c860472722f98f36c164, Version: d8cfedd31e23b815ad861b2e68523a31664eaf28c2ddc7b682eb115646a473d9
```

Comment (penandlim) [reply src/cache.js:31]: https://github.com/actions/toolkit/blob/415c42d27ca2a24f3801dd9406344aaea00b7866/packages/cache/src/cache.ts#L245-L246

**Review by onbjerg — APPROVED** [review]
thank you

**Pull request merged**